pkg/utils/tokens.go

```diff
@@ -0,0 +1,89 @@
+package utils
+
+import (
+	"math/big"
+	"sort"
+)
+
+type Partitioner struct {
+	RingOffset *big.Int
+	RingRange  *big.Int
+}
+
+var one = bigInt(1)
+
+var Murmur3Partitioner = Partitioner{
+	RingOffset: new(big.Int).Lsh(bigInt(-1), 63), // -1<<63
+	RingRange:  new(big.Int).Lsh(one, 64),        // 1<<64
+}
+
+var RandomPartitioner = Partitioner{
+	RingOffset: new(big.Int),               // 0
+	RingRange:  new(big.Int).Lsh(one, 127), // 1<<127
+}
+
+// ComputeTokens takes a list of numbers indicating the number of nodes in each datacenter, and returns a recommended
+// list of tokens for the given partitioner (one token per node, grouped by datacenter).
+func ComputeTokens(dcCounts []int, partitioner Partitioner) [][]string {
```

Comment:
Returning strings because I think that will be the most convenient in the calling code. If we prefer raw numerical values it's an easy change.